packages/camera/camera_windows/example/lib/main.dart

```diff
@@ -0,0 +1,460 @@
+}
+
+class _MyAppState extends State<MyApp> {
+  String _cameraInfo = 'Unknown';
+  List<CameraDescription> _cameras = <CameraDescription>[];
+  int _cameraIndex = 0;
+  int _cameraId = -1;
+  bool _initialized = false;
+  bool _recording = false;
+  bool _recordingTimed = false;
+  bool _recordAudio = true;
+  bool _previewPaused = false;
+  Size? _previewSize;
+  ResolutionPreset _resolutionPreset = ResolutionPreset.veryHigh;
+  StreamSubscription<CameraErrorEvent>? _errorStreamSubscription;
+  StreamSubscription<CameraClosingEvent>? _cameraClosingStreamSubscription;
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsFlutterBinding.ensureInitialized();
+    _fetchCameras();
+  }
+
+  @override
+  void dispose() {
+    _disposeCurrentCamera();
+    _errorStreamSubscription?.cancel();
+    _errorStreamSubscription = null;
+    _cameraClosingStreamSubscription?.cancel();
+    _cameraClosingStreamSubscription = null;
+    super.dispose();
+  }
+
+  /// Fetches list of available cameras from camera_windows plugin.
+  Future<void> _fetchCameras() async {
+    String cameraInfo;
+    List<CameraDescription> cameras = <CameraDescription>[];
+
+    int cameraIndex = 0;
+    try {
+      cameras = await CameraPlatform.instance.availableCameras();
+      if (cameras.isEmpty) {
+        cameraInfo = 'No available cameras';
+      } else {
+        cameraIndex = _cameraIndex % cameras.length;
+        cameraInfo = 'Found camera: ${cameras[cameraIndex].name}';
+      }
+    } on PlatformException catch (e) {
+      cameraInfo = 'Failed to get cameras: ${e.code}: ${e.message}';
+    }
+
+    if (mounted) {
+      setState(() {
+        _cameraIndex = cameraIndex;
+        _cameras = cameras;
+        _cameraInfo = cameraInfo;
+      });
+    }
+  }
+
+  /// Initializes the camera on the device.
+  Future<void> _initializeCamera() async {
+    assert(!_initialized);
+
+    if (_cameras.isEmpty) {
+      return;
+    }
+
+    int cameraId = -1;
+    try {
+      final Completer<CameraInitializedEvent> _initializeCompleter =
+          Completer<CameraInitializedEvent>();
+
+      final int cameraIndex = _cameraIndex % _cameras.length;
+      final CameraDescription camera = _cameras[cameraIndex];
+
+      cameraId = await CameraPlatform.instance.createCamera(
+        camera,
+        _resolutionPreset,
+        enableAudio: _recordAudio,
+      );
+
+      _errorStreamSubscription?.cancel();
+      _errorStreamSubscription = CameraPlatform.instance
+          .onCameraError(cameraId)
+          .listen(_onCameraError);
+
+      _cameraClosingStreamSubscription?.cancel();
+      _cameraClosingStreamSubscription = CameraPlatform.instance
+          .onCameraClosing(cameraId)
+          .listen(_onCameraClosing);
+
+      unawaited(CameraPlatform.instance
+          .onCameraInitialized(cameraId)
+          .first
+          .then((CameraInitializedEvent event) {
+        _initializeCompleter.complete(event);
+      }));
```

Comment:
You can avoid using a `Completer` here by doing:
```dart
Future initialized = CameraPlatform.instance
    .onCameraInitialized(cameraId)
    .first;

...

CameraInitializedEvent event = await initialized;
_previewSize = Size(event.previewWidth, event.previewHeight);
```

Reply:
Yes, this is much better approach... 
This should be changed at camera plugin as well: [reference](https://github.com/flutter/plugins/blob/main/packages/camera/camera/lib/src/camera_controller.dart#L280-L325)
Fixed and working.